feat: add docker plugin install (#647)
zwave-js/zwavejs2mqtt: docs/guide/plugins.md

# Plugins

> Plugins are now supported in the application.

Reviewer: I would add some more descriptive comment here. Something like: Plugins are nodejs applications that can be initegrated with the application in order to add some features. They have access to all the clients and express instance

In order to support these, we need:

> 1. Build docker with the plugin

Reviewer: I would remove this, in theory you could use them with all the others ways supported, you don't need docker

2. Configure our settings.json (manually)

Building the container is straight forward. an example of build command installing two plugins `@varet/zj2m-prom-exporter`

```bash
TAG="v1.1.1-test"; docker build -f docker/Dockerfile --build-arg plugins="@varet/zj2m-prom-exporter" -t localdocker.dc.rb11.eu/zwavejs2mqtt:${TAG} .
```

Once container is built we need to configure our `store/settings.json`

An example of valid config is:

> ```json
> {
>   "mqtt": {
>     "name": "home",
>     ...
>     "store": false
>   },
>   "gateway": {
>     "type": 0,
>     ...
>     "sendEvents": false
>   },
>   "zwave": {
>     "port": "/dev/tty.usbserial",
>     ...
>     "plugin": "@varet/zj2m-prom-exporter"
>   }
> }
> ```

Reviewer: ```suggestion
```json
{
  "mqtt": {
    ...
  },
  "gateway": {
    ...
  },
  "zwave": {
    ...
    "plugin": "@varet/zj2m-prom-exporter"
  }
}
```
```

# Plugins

> Plugins are now supported in the application. The plugin packages are not included when npm is installing packages

Reviewer: ```suggestion
Plugins are nodejs packages that can be initegrated in zwavejsmqtt in order to add new features. They are have access to all the clients (zwave and mqtt) and express instance.
```

> ## Installing Plugin

Reviewer: ```suggestion
## Usage
```

> ### NPM Method

Reviewer: ```suggestion
### Installation
```

> you can install a plugin using `npm i` (install) command. It is advised to run this command after building.

Reviewer: ```suggestion
The plugin is imported in zwavejs1mqtt using `require(pluginName)(zwave, mqtt, app)` 

- `zwave`: Zwave client
- `mqtt`: Mqtt client
- `app`: Express instance
```

example installing `@user/plugin-name`

```bash
npm i @user/plugin-name

> ```

Reviewer: When using plugins you can place them in a custom folder and use as plugin name the absolute path to it. Otherwise if them are published to npm you can more easily run `npm install <pluginName>` and use just `<pluginName>` in settings to use them instead of specify the absolute path.

> ### Building it using docker file

Reviewer: ```suggestion
### Plugins with docker
```

> Building the container is straight forward. an example of build command installing two plugins `@user/plugin-name`

Reviewer: ```suggestion
Building the container is straight forward. Here an example of build command installing plugin `path/to/plugin1`
```

```bash

> docker build -f docker/Dockerfile --build-arg plugins='@user/plugin-name' -t <docker image name>:<tag> .

Reviewer: ```suggestion
docker build -f docker/Dockerfile --build-arg plugins='path/to/plugin1' -t <docker image name>:<tag> .
```

```

Once container is built we need to configure our `store/settings.json`

An example of valid config is:

```json
{
  "mqtt": {
    ...
  },
  "gateway": {
    ...
  },
  "zwave": {
    ...

>     "plugin": "@user/plugin-name"

Reviewer: ```suggestion
    "plugin": "path/to/plugin1"
```
